AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/fragment/StreamConfigurationFragment.java

```diff
@@ -72,12 +73,22 @@ public View onCreateView(final LayoutInflater inflater,
 
         final View view = inflater.inflate(R.layout.fragment_stream_configuration, container, false);
 
+        final Thread thread = new Thread(() -> {
+            try {
+                ((AWSMobileClient) KinesisVideoDemoApp.getCredentialsProvider()).getAWSCredentials();
+            } catch (final Exception e) {
+                Log.e(TAG, "Exception while fetching credentials", e);
```

Comment:
Why are we fetching the credentials again here?

Reply:
We fetch the credentials before we create the Kinesis video client. Otherwise we run into
`com.amazonaws.services.cognitoidentity.model.NotAuthorizedException: Access to Identity 'us-west-2:x' is forbidden. (Service: null; Status Code: 400; Error Code: NotAuthorizedException; Request ID: x)
E/KinesisVideoAndroidClient: 2022-02-14T11:56Z T2: EXCEPTION: ExecutionException: Awaiting for the credentials update threw an exception: com.amazonaws.services.cognitoidentity.model.NotAuthorizedException: Access to Identity 'us-west-2:x' is forbidden. (Service: AmazonCognitoIdentity; Status Code: 400; Error Code: NotAuthorizedException; Request ID: x)`

Reply:
Do we need to authenticate later again? Have you tried creating the client without creds below since we already authenticated once?

Reply:
Confirmed that credentials are being cached: 
`CognitoCachingCredentialsProvider cognitoIdentity;`

Reply:
Discussed offline. The [source code](https://github.com/aws-amplify/aws-sdk-android/blob/main/aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java#L219) uses a caching creds provider.